homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,233 @@
+"""Config flow to configure Amcrest devices."""
+
+import logging
+
+from amcrest import AmcrestError, LoginError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_BINARY_SENSORS,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_SENSORS,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from .amcrest_checker import AmcrestChecker
+from .const import (
+    AUTHENTICATION_LIST,
+    CONF_BINARY_SENSOR_AUDIO_DETECTED,
+    CONF_BINARY_SENSOR_AUDIO_DETECTED_POLLED,
+    CONF_BINARY_SENSOR_MOTION_DETECTED,
+    CONF_BINARY_SENSOR_MOTION_DETECTED_POLLED,
+    CONF_BINARY_SENSOR_ONLINE,
+    CONF_CONTROL_LIGHT,
+    CONF_EVENTS,
+    CONF_FFMPEG_ARGUMENTS,
+    CONF_RESOLUTION,
+    CONF_SENSOR_PTZ_PRESET,
+    CONF_SENSOR_SDCARD,
+    CONF_STREAM_SOURCE,
+    DEFAULT_AUTHENTICATION,
+    DEFAULT_CONTROL_LIGHT,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_RESOLUTION,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_STREAM_SOURCE,
+    DEVICES,
+    DOMAIN,
+    RESOLUTION_LIST,
+    STREAM_SOURCE_LIST,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AmcrestFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
```

Comment:
Where are the tests of the config flow?

Reply:
What is missing?
I have tested it and it works

Reply:
my bad, i have read "the rest" instead of "the tests"
now I create the tests

Reply:
I have a problem with creating the test.
i get the error
"TypeError: Object of type MagicMock is not JSON Serializable"
on the call 
``
with patch(
        "homeassistant.components.amcrest.config_flow.AmcrestChecker.serial_number",
        return_value=TEST_SERIALNUMBER,
    ):
``


it appears that a MagicMock object is returned instead of a String

but I don't understand where I'm wrong

Reply:
test added

Reply:
test added